### DISCUSSION TITLE
Should target face diameter be in different units?

### DISCUSSION THREAD

**Discussion by jatkinson1000** [jatkinson1000/archeryutils#17] [Ideas]
At present target face diameter is set in metres.
This is to keep things in SI units and try to negate bugs.

However, there is an argument that it would be more intuitive for users to set in cm.
And to perhaps add the option of setting in inches as well.

Thoughts?
Based on discussion this could become an issue with a feature request.

**Comment by jatkinson1000** [comment]
This would also allow for improvements to the `get_info` method in `Round` which currently only outputs face diameter in cm.

**Comment by LiamPattinson** [comment]
People normally quote target sizes in cm instead of m, so I think that'd be more intuitive. It also means you can specify sizes exactly as an int and avoid floating-point round off, but I really doubt that matters in practice.

One option could be to have an optional keyword argument of `units='cm'` so users can choose their input units and the `__init__` function would do the conversion for them. A more heavyweight option would be to bring in something like [pint](https://pint.readthedocs.io/en/stable/) which you can use to convert units on function entry, but that might be overkill.

Also, which rounds use targets sized in inches? I'd guess that 122cm faces used to be sized in inches since it's approximately 4 feet, but I wasn't aware of anybody still doing it that way.

**Comment by jatkinson1000** [comment]
Yeah - indeed I've been caught out a number of times when using this because I intuitively use the cm measurement, and I'm the one who wrote it!!
The one thing I don't like is you'd then have to put a factor of 100 in all of the arrow score calculations which my physical science brain strongly dislikes.
One way around could be to put the factor of 100 in the setter for target diameter so users supply it in cm, but the code stores the value in metres?

I did consider pint, but at the time of writing I had too many other bits to sort to think more about it.

EFAA still uses inches, but even within AGB the Worcester is still in inches.

**Comment by jatkinson1000** [comment]
I think best approach is to provide tooling for users to specify in cm, inches, or m, with the default being cm, but they store everything under the hood in m so as to preserve SI units and make the code less prone to errors. I'll open an issue requesting this.

**Comment by jatkinson1000** [comment]
#27 Opened to reflect this.
